apps/src/turtle/customLevelBlocks.js

```diff
@@ -39,28 +39,80 @@ function createACircleCode (size, gensym, indent) {
     indent + '}\n'].join('\n');
 }
 
-
 /**
- * Same as draw_a_square, except inputs are not inlined
+ * Returns an initialization object that sets up blockly attributes;
+ *
+ * @param title - The title of the block that will be visible to the user.
+ * @param [argument] - Optional parameter for blocks that accept a value
+ *    argument. This is the title of the argument.
+ * @param [titleValue] - Optional parameter to associate a value with the
+ *    title.
+ * @return the initialization object
  */
-function installDrawASquare(blockly, generator, gensym) {
-  // Create a fake "draw a square" function so it can be made available to users
-  // without being shown in the workspace.
-  blockly.Blocks.draw_a_square_custom = {
-    // Draw a square.
+function makeInitializer(title, argument, titleValue) {
+  return {
     init: function() {
       this.setHSV(94, 0.84, 0.60);
-      this.appendDummyInput()
-          .appendTitle(msg.drawASquare());
-      this.appendValueInput('VALUE')
-          .setAlign(blockly.ALIGN_RIGHT)
-          .setCheck(Blockly.BlockValueType.NUMBER)
-              .appendTitle(msg.lengthParameter() + ':');
+
+      this.appendDummyInput().appendTitle(title, titleValue);
+
+      if (argument !== undefined) {
+        this.appendValueInput('VALUE')
+            .setAlign(Blockly.ALIGN_RIGHT)
+            .setCheck(Blockly.BlockValueType.NUMBER)
+            .appendTitle(argument + ':');
+      }
+
       this.setPreviousStatement(true);
       this.setNextStatement(true);
       this.setTooltip('');
     }
   };
+}
```

Comment:
Heh, yea I went through multiple variants naming that. I settled on `makeInitializer` because the lines were reading:

**block**ly.**Block**s.draw_a_square_custom = make**Block**Initializer(title, LENGTH_PARAM);

In context it seemed apparent that it was for blocks since 'block' was already on each line twice, and the function itself doc'd it's purpose. I've since changed this to use your suggestion.

Re: makeInitializerWithAssociatedTitleValue - interestingly enough I had to rip this out and do an explicit impl for that initializer any way. See that line for more context.

I also did s/argument/parameter/ on my methods.